src/pytest_cpp/boost.py

```diff
@@ -30,7 +39,12 @@ def is_test_suite(cls, executable: str) -> bool:
         else:
             return "--output_format" in output and "log_format" in output
 
-    def list_tests(self, executable: str) -> list[str]:
+    def list_tests(
+        self,
+        executable: str,
+        prefix: Sequence[str] = (),
```

Comment:
Not sure if it is better to remove the default value for `prefix`... we probably want a hard exception in case we do not pass it explicitly.

Reply:
I do believe exception is not needed here because prefix cpp_harness_collect might be ommited

Reply:
Not sure I follow... I just meant to remove the default value, so if somebody forgets to pass it Python itself will raise an exception. This is used internally only, so it is best to ensure we are not forgetting to pass `prefix` at some point (and having a default value would let that error slip by).

Reply:
I did try it, however this require a lot of changes in exist tests because I do not see any reason to remove a default value for a `prefix` in the observation/collecting functions and keep it for a `harness` in a test run, and to be able to do that I need to reorder a list of arguments for `run_test`.

Reply:
OK